robots/pkg/kubevirtci/kubevirtci_test.go

```diff
@@ -334,3 +335,191 @@ func release(tagName string, released bool) *github.RepositoryRelease {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "no supported releases provokes error",
+			args: args{
+				scenario: testScenario{
+					dirsBefore: []string{
+						"cluster-provision/k8s/1.19",
+						"cluster-provision/k8s/1.20",
+						"cluster-provision/k8s/1.21",
+						"cluster-up/cluster/k8s-1.19",
+						"cluster-up/cluster/k8s-1.20",
+						"cluster-up/cluster/k8s-1.21",
+					},
+				},
+				providerDir:       "cluster-provision/k8s",
+				clusterUpDir:      "cluster-up/cluster",
+				supportedReleases: []*github.RepositoryRelease{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "only supported providers left",
+			args: args{
+				scenario: testScenario{
+					dirsBefore: []string{
+						"cluster-provision/k8s/1.19",
+						"cluster-provision/k8s/1.20",
+						"cluster-provision/k8s/1.21",
+						"cluster-up/cluster/k8s-1.19",
+						"cluster-up/cluster/k8s-1.20",
+						"cluster-up/cluster/k8s-1.21",
+					},
+					providerDirsAfter: []string{
+						"1.19",
+						"1.20",
+						"1.21",
+					},
+					clusterUpDirsAfter: []string{
+						"k8s-1.19",
+						"k8s-1.20",
+						"k8s-1.21",
+					},
+				},
+				providerDir:  "cluster-provision/k8s",
+				clusterUpDir: "cluster-up/cluster",
+				supportedReleases: []*github.RepositoryRelease{
+					{
+						TagName: strPointer("v1.19.3"),
+					},
+					{
+						TagName: strPointer("v1.20.2"),
+					},
+					{
+						TagName: strPointer("v1.21.1"),
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "unsupported providers should get deleted",
+			args: args{
+				scenario: testScenario{
+					dirsBefore: []string{
+						"cluster-provision/k8s/1.19",
+						"cluster-provision/k8s/1.20",
+						"cluster-provision/k8s/1.21",
+						"cluster-up/cluster/k8s-1.19",
+						"cluster-up/cluster/k8s-1.20",
+						"cluster-up/cluster/k8s-1.21",
+					},
+					providerDirsAfter: []string{
+						"1.20",
+						"1.21",
+					},
+					clusterUpDirsAfter: []string{
+						"k8s-1.20",
+						"k8s-1.21",
+					},
+				},
+				providerDir:  "cluster-provision/k8s",
+				clusterUpDir: "cluster-up/cluster",
+				supportedReleases: []*github.RepositoryRelease{
+					{
+						TagName: strPointer("v1.20.2"),
+					},
+					{
+						TagName: strPointer("v1.21.1"),
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tempDir, err := os.MkdirTemp("", "dropProviders-")
+			panicOnErr(err)
+			defer os.RemoveAll(tempDir)
+			for _, dir := range tt.args.scenario.dirsBefore {
+				panicOnErr(mkDirs(path.Join(tempDir, dir)))
+			}
+
+			targetClusterUpDir := path.Join(tempDir, tt.args.clusterUpDir)
+			targetProviderDir := path.Join(tempDir, tt.args.providerDir)
+			err = DropUnsupportedProviders(targetProviderDir, targetClusterUpDir, tt.args.supportedReleases)
+			if (err != nil) != tt.wantErr {
+				t.Fatalf("DropUnsupportedProviders() error = %v, wantErr %v", err, tt.wantErr)
+				return
```

Comment:
/lgtm
/approve
/hold

With `t.Fatalf` we don't need the `return`, but feel free to ignore, can be addressed later :)

Reply:
:facepalm: removed. Thanks!